include/StochasticNoiseSimulator.hpp

```diff
@@ -59,33 +65,79 @@ class StochasticNoiseSimulator: public Simulator {
 
     [[nodiscard]] std::size_t getNumberOfOps() const override { return qc->getNops(); };
 
-    [[nodiscard]] std::string getName() const override { return "stoch_" + gate_noise_types + "_" + qc->getName(); };
+    [[nodiscard]] std::string getName() const override { return "stoch_" + gateNoiseEffects + "_" + qc->getName(); };
 
-    void setNoiseEffects(const std::string& cGateNoise) { gate_noise_types = cGateNoise; }
+    double           noiseProbability = 0.0;
+    dd::ComplexValue sqrtAmplitudeDampingProbability{};
+    dd::ComplexValue oneMinusSqrtAmplitudeDampingProbability{};
 
-    double           noise_probability = 0.0;
-    dd::ComplexValue sqrt_amplitude_damping_probability{};
-    dd::ComplexValue one_minus_sqrt_amplitude_damping_probability{};
+    double           noiseProbabilityMulti = 0.0;
+    dd::ComplexValue sqrtAmplitudeDampingProbabilityMulti{};
+    dd::ComplexValue oneMinusSqrtAmplitudeDampingProbabilityMulti{};
 
-    void setAmplitudeDampingProbability(double cGateNoiseProbability) {
+    dd::GateMatrix ampDampingTrue      = {};
+    dd::GateMatrix ampDampingTrueMulti = {};
+
+    dd::GateMatrix ampDampingFalse      = {};
+    dd::GateMatrix ampDampingFalseMulti = {};
+
+    void setNumberOfRuns(long stoch_run) {
+        if (stoch_run <= 0) {
+            throw std::runtime_error("Number of stochastic runs must be larger than 0. Provided value: " + std::to_string(stoch_run));
+        }
+        stochasticRuns = stoch_run;
+    }
```

Comment:
Make parameter `unsigned int` or better yet `std::size_t` (same for `stochasticRuns`).

Also camelCase.